backend/src/main/scala/com/softwaremill/bootzooka/util/SecureRandomId.scala

```diff
@@ -0,0 +1,48 @@
+package com.softwaremill.bootzooka.util
+
+import cats.effect.Sync
+import java.security.SecureRandom
+import org.apache.commons.codec.binary.Hex
+
+/** copy-pasted from TSec https://github.com/jmcardon/tsec */
+
+/** A trait that manages a secureRandom instance.
+  */
+trait ManagedRandom {
+
+  /** Cache our random, and seed it properly as per
+    * [[https://tersesystems.com/2015/12/17/the-right-way-to-use-securerandom/]]
+    */
+  private val cachedRand: SecureRandom = {
+    val r =
+      SecureRandom.getInstance(if (scala.util.Properties.isWin) ManagedRandom.WinRandom else ManagedRandom.UnixURandom)
+    r.nextBytes(new Array[Byte](20)) //Force reseed
+    r
+  }
+
+  def nextBytes(bytes: Array[Byte]): Unit =
+    cachedRand.nextBytes(bytes)
+}
+
+object ManagedRandom {
+  private[ManagedRandom] val WinRandom   = "Windows-PRNG"
+  private[ManagedRandom] val UnixURandom = "NativePRNGNonBlocking"
+}
+
+object SecureRandomId {
+  lazy val Strong: SecureRandomIdGenerator      = SecureRandomIdGenerator(32)
+  lazy val Interactive: SecureRandomIdGenerator = SecureRandomIdGenerator(16)
+
+  def apply(s: String): SecureRandomId  = s.asInstanceOf[SecureRandomId]
+  def coerce(s: String): SecureRandomId = s.asInstanceOf[SecureRandomId]
```

Comment:
do we need both apply and coerce?

Reply:
no, `coerce` is not used, so it would be removed